src/Discord.Net.Rest/API/Common/ApplicationCommandInteractionData.cs

```diff
@@ -0,0 +1,23 @@
+using Newtonsoft.Json;
+
+namespace Discord.API
+{
+    [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
+    internal class ApplicationCommandInteractionData
```

Comment:
duplicate of https://github.com/Discord-Net-Labs/Discord.Net-Labs/blob/Interactions/src/Discord.Net.Rest/API/Common/ApplicationCommandInteractionData.cs

The way the IDiscordInteractionData deserialization works uses the interface for it.